### PR TITLE
Reorder documentation

### PR DESCRIPTION
**Pull request by MagdalenaZuba**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   |https://issues.ibexa.co/browse/IBX-5860 and https://issues.ibexa.co/browse/IBX-5974
| Versions      | 4.6
| Edition       | Commerce

#### Checklist

- [ ] Text renders correctly
- [ ] Text has been checked with vale
- [ ] Description metadata is up to date
- [ ] Redirects cover removed/moved pages
- [ ] Code samples are working
- [ ] PHP code samples have been fixed with PHP CS fixer
- [ ] Added link to this PR in relevant JIRA ticket or code PR

### CODE REVIEW

File: code_samples/api/commerce/src/Command/CartCommand.php [hunk 1]
```diff
@@ -135,5 +135,18 @@ protected function execute(InputInterface $input, OutputInterface $output): int
         $this->cartService->deleteCart($cart);
 
         return self::SUCCESS;
+
+        // Get the target cart
```
Comment (ViniTou) [inline code_samples/api/commerce/src/Command/CartCommand.php:139]: This is not target cart, this is order with items that should be reordered

File: code_samples/api/commerce/src/Command/CartCommand.php [hunk 2]
```diff
@@ -135,5 +135,18 @@ protected function execute(InputInterface $input, OutputInterface $output): int
...
+        $order = $this->orderService->getOrderByIdentifier($orderIdentifier);
+
+        // Get the carts to merge
```
Comment (ViniTou) [inline code_samples/api/commerce/src/Command/CartCommand.php:142]:
```suggestion
        // Get the cart to merge
```

File: docs/commerce/checkout/reorder.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+                        ...
+                        can_be_reordered: true
+                !php/const Ibexa\OrderManagement\Value\Status::CANCELLED_PLACE:
+                    metadata:
+                        ...
+                        can_be_reordered: true
+
+```
+
+The next example shows how to hide a step during checkout in reorder workflow.
```
Comment (ViniTou) [inline docs/commerce/checkout/reorder.md:66]: Not sure that this should be documented here, as it is not related to reorder process, even more it is not used in current configuration.

File: docs/commerce/checkout/reorder.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+```
+
+## Reorder PHP API
+
+You can manage and modify reorder with a dedicated checkout and cart PHP API.
+
+### Checkout PHP API
+
+Reorder comes with dedicated `Ibexa\Contracts\Checkout\Reorder\ReorderService` interface.
+You can use it to manage reorders in your project.
```
Comment (ViniTou) [inline docs/commerce/checkout/reorder.md:90]: There is no managing when it comes to reorders, those are just normal orders and are by that you use OrderService to manipulate them. 
ReorderService just have some helpers method/facades over existing API to ease the process.

File: code_samples/api/commerce/src/Command/CartCommand.php [hunk 5]
```diff
@@ -135,5 +135,18 @@ protected function execute(InputInterface $input, OutputInterface $output): int
...
+
+        // Get the cart to merge
+        $existingCart = $this->cartResolver->resolveCart();
+
+        $reorderCart = $this
+            ->reorderService
+            ->addToCartFromOrder($order, $this->reorderService->createReorderCart($order));
+
+        // Merge the carts into the target cart and delete the merged carts
+        $reorderCart = $this->cartService->mergeCarts($reorderCart, true, $existingCart);
```
Comment (ViniTou) [inline code_samples/api/commerce/src/Command/CartCommand.php:150]: just FYI, this part is optional, it is perfectly valid to checkout with just the reorderCart without merging it with existing one.

**Review by ViniTou — APPROVED** [review]

File: docs/commerce/checkout/reorder.md [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+### Checkout PHP API
+
+Reorder comes with dedicated `Ibexa\Contracts\Checkout\Reorder\ReorderService` interface.
+It contains helper methods and facades added over existing API to ease the order manipulation process.
+The following methods can be used to modify the reorder flow to fit your business needs:
+
+#### `ReorderService:addToCartFromOrder` 
+
+Allows you to add items from a previous order to a cart.
+It uses historic data from previously ordered items even if they are no longer available.
```
Comment (juskora) [inline docs/commerce/checkout/reorder.md:79]:
```suggestion
It uses historical data from previously ordered items even if they are no longer available.
```

File: docs/commerce/order_management/configure_order_management.md [hunk 7]
```diff
@@ -43,6 +43,20 @@ Then reference it with `ibexa.repositories.<your_repository>.order_management.wo
 [[= include_file('code_samples/front/shop/order-management/config/packages/ibexa.yaml', 69, 74) =]]
 ```
 
+### Define cancel order
+
+You can define a status and transition in which the order can be cancelled
```
Comment (juskora) [inline docs/commerce/order_management/configure_order_management.md:48]:
```suggestion
You can define a status and transition in which the order can be canceled 
```

File: docs/commerce/storefront/storefront.md [hunk 8]
```diff
@@ -16,17 +16,21 @@ You can modify them when you build your own web store.
 
 | Component | Description |
 |------------|----------|
+|Cancel order|Allows logged in users to cancel their orders in a pending status.|
```
Comment (juskora) [inline docs/commerce/storefront/storefront.md:19]:
```suggestion
|Cancel order|Allows logged-in users to cancel their orders in a pending status.|
```

File: docs/commerce/storefront/storefront.md [hunk 9]
```diff
@@ -16,17 +16,21 @@ You can modify them when you build your own web store.
...
 | Language menu | Enables selecting between languages, to change an active language. |
 | Login/register page |  Provides user interface for the login/registration page that enables buyers to access the Product catalog.|
 | Main cart component | Main UI component of the cart. Displays a list of items selected for purchase and requested cart item quantities. Users can remove individual items. |
 | Mini cart widget | Consists of a counter that displays a total number of items added to a cart. |
+|Orders list|Displays a list of orders with such information as status, date, value, order ID. |
 | Product category page | Displays products that belong to a specific category. |
 | Product filters component | Allows for narrowing the list of products displayed in the listing by using different filters, such as product type, availability and price. |
 | Product listing page | Allows for browsing through products, displays product name, code, price, and image. |
 | Region menu | Enables selecting between regions, to dynamically change the contents of the product listing page. | 
+| Reorder |Allows logged in users to repurchase previously bought items. |
```
Comment (juskora) [inline docs/commerce/storefront/storefront.md:32]:
```suggestion
| Reorder |Allows logged-in users to repurchase previously bought items. |
```

File: docs/commerce/storefront/storefront.md [hunk 10]
```diff
@@ -16,17 +16,21 @@ You can modify them when you build your own web store.
...
+|Searching and filtering of orders| Allows logged in users to search and filter their past orders on the orders page.|
```
Comment (juskora) [inline docs/commerce/storefront/storefront.md:33]:
```suggestion
|Searching and filtering of orders| Allows logged-in users to search and filter their past orders on the orders page.|
```

**Review by juskora — CHANGES_REQUESTED** [review]

**Review by juskora — APPROVED** [review]

**Review by radzyu — APPROVED** [review]
Great work @MagdalenaZuba, thank you!

**Pull request merged**